common/workunit/workflow.cpp

```diff
@@ -613,12 +808,888 @@ WorkflowMachine::WorkflowMachine(const IContextLogger &_logctx)
 {
 }
 
+
+void WorkflowMachine::addSuccessors()
+{
+    Owned<IRuntimeWorkflowItemIterator> iter = workflow->getSequenceIterator();
+    if (iter->first())
+    {
+        while (iter->isValid())
+        {
+            IConstWorkflowItem * item = iter->query();
+            if(item->queryState() == WFStateReqd)
+            {
+                //initial call
+                startingWfid = item->queryWfid();
+#ifdef TRACE_WORKFLOW
+                LOG(MCworkflow, "Item %u has been identified as the 'parent' item, with Reqd state", startingWfid);
+#endif
+                markDependencies(startingWfid, nullptr, false);
+                break;
+            }
+            if(!iter->next()) break;
+        }
+    }
+#ifdef TRACE_WORKFLOW
+    //Outputting debug info about each workflow item.
+    unsigned totalDependencies = 0;
+    unsigned totalActiveItems = 0;
+    unsigned totalInActiveItems = 0;
+    unsigned totalDependentSuccessors = 0;
+    unsigned totalLogicalSuccessors = 0;
+    unsigned totalConditionItems = 0;
+    //iterate through the workflow items
+    for(int i = 1; i <= workflow->count(); i++)
+    {
+        CCloneWorkflowItem & cur = queryWorkflowItem(i);
+        unsigned numDep = cur.queryNumDependencies();
+        unsigned numDepSuc = cur.queryNumDependentSuccessors();
+        unsigned numLogSuc = cur.queryNumLogicalSuccessors();
+        if(cur.isActive())
+            totalActiveItems ++;
+        else
+            totalInActiveItems ++;
+        totalDependencies += numDep;
+        totalDependentSuccessors += numDepSuc;
+        totalLogicalSuccessors += numLogSuc;
+        LOG(MCworkflow, "Item %u has %u dependencies, %u dependent successors and %u logical successors", cur.queryWfid(), numDep, numDepSuc, numLogSuc);
+
+        if(cur.queryMode() == WFModeCondition)
+        {
+            totalConditionItems++;
+        }
+    }
+    //iterate throught the IntermediaryWorkflow items
+    for(int i = 0;  i < logicalWorkflow.size() ; i++)
+    {
+
+        IRuntimeWorkflowItem  *tmp = logicalWorkflow[i].get();
+        CCloneWorkflowItem * cur = static_cast<CCloneWorkflowItem*>(tmp);
+        unsigned numDep = cur->queryNumDependencies();
+        unsigned numDepSuc = cur->queryNumDependentSuccessors();
+        unsigned numLogSuc = cur->queryNumLogicalSuccessors();
+        if(cur->isActive())
+            totalActiveItems ++;
```

Comment:
trivial: don't normally have space between the item and ++.  (same for a few other lines in this function)